flycheck.el

```diff
@@ -6814,7 +6814,13 @@ See URL `https://github.com/commercialhaskell/stack'."
   (lambda (errors)
     (flycheck-sanitize-errors (flycheck-dedent-error-messages errors)))
   :modes (haskell-mode literate-haskell-mode)
-  :next-checkers ((warning . haskell-hlint)))
+  :next-checkers ((warning . haskell-hlint))
+  :predicate (lambda ()
+               (let ((fs (mapcar (lambda (f)
+					(and f (locate-dominating-file (buffer-file-name) f)))
```

Comment:
Since you're using `(buffer-file-name)` at this place, the `:predicate` also needs to ensure that `(buffer-file-name)` is `non-nil`.